internal/pkg/vault/management.go

```diff
@@ -247,3 +247,275 @@ func (c *Client) ConfigureConsulAccess(secretStoreToken string, bootstrapACLToke
+
+	urlPath := path.Join(namedEntityAPI, name)
+	response := ReadEntityByNameResponse{}
+
+	code, err := c.doRequest(RequestArgs{
+		AuthToken:            secretStoreToken,
+		Method:               http.MethodGet,
+		Path:                 urlPath,
+		JSONObject:           nil,
+		BodyReader:           nil,
+		OperationDescription: "Read Entity",
+		ExpectedStatusCode:   http.StatusOK,
+		ResponseObject:       &response,
+	})
+
+	// 404 is a valid response, either if there are no entities at all
+	// or if the one we are looking for isn't there
+
+	if code == http.StatusNotFound {
+		return "", nil
+	} else if err != nil {
+		return "", err
+	}
+
+	return response.Data.ID, nil
+}
+
+func (c *Client) CheckAuthMethodEnabled(token string, mountPoint string, authType string) (bool, error) {
+	var response ListSecretEnginesResponse // Coincidentially the same as the sys/auth output
+
+	_, err := c.doRequest(RequestArgs{
+		AuthToken:            token,
+		Method:               http.MethodGet,
+		Path:                 authAPI,
+		JSONObject:           nil,
+		BodyReader:           nil,
+		OperationDescription: "query auth engine for" + authType,
+		ExpectedStatusCode:   http.StatusOK,
+		ResponseObject:       &response,
+	})
+
+	if err != nil {
+		return false, err
+	}
+
+	if mountData := response.Data[mountPoint]; mountData.Type == authType {
+		return true, nil
+	}
+
+	return false, nil
+}
+
+func (c *Client) EnablePasswordAuth(token string, mountPoint string) error {
+	urlPath := path.Join(authAPI, mountPoint)
+	parameters := EnableAuthMethodRequest{
+		Type: UsernamePasswordAuthMethod,
+	}
+
+	_, err := c.doRequest(RequestArgs{
+		AuthToken:            token,
+		Method:               http.MethodPost,
+		Path:                 urlPath,
+		JSONObject:           parameters,
+		BodyReader:           nil,
+		OperationDescription: "Enable password auth",
+		ExpectedStatusCode:   http.StatusNoContent,
+		ResponseObject:       nil,
+	})
+
+	return err
+}
+
+func (c *Client) LookupAuthHandle(token string, mountPoint string) (string, error) {
+	response := ListAuthMethodsResponse{}
+
+	_, err := c.doRequest(RequestArgs{
+		AuthToken:            token,
+		Method:               http.MethodGet,
+		Path:                 authAPI,
+		JSONObject:           nil,
+		BodyReader:           nil,
+		OperationDescription: "List auth methods",
+		ExpectedStatusCode:   http.StatusOK,
+		ResponseObject:       &response,
+	})
+
+	if err != nil {
+		return "", err
+	}
+
+	info, ok := response.Data[mountPoint+"/"]
+	if !ok {
+		return "", fmt.Errorf("Mount point %s not found", mountPoint)
+	}
+	return info.Accessor, nil
+}
+
+// FIXME: separate identity policies from token policies
+// see https://groups.google.com/g/vault-tool/c/xyhIAC44RmI?pli=1
+func (c *Client) CreateOrUpdateUser(token string, mountPoint string, username string, password string, tokenTTL string, tokenPolicies []string) error {
+	urlPath := path.Join(authMountBase, mountPoint, "users", username)
```

Comment:
suggest do `username` input validation before creating urlPath using it

Reply:
Validation added.